src/message/connection.js

```diff
@@ -531,7 +531,8 @@ Connection.prototype._tryReconnect = function () {
   } else {
     this._clearReconnect()
     this.close()
-    this._client.emit(C.MAX_RECONNECTION_ATTEMPTS_REACHED, this._reconnectionAttempt)
+    this._setState(C.CONNECTION_STATE.CLOSED)
```

Comment:
I'd like to get this into a release I'm doing today so if you'd could remove the setState call that would be great.
From looking at the code I can't see what would put the connection state into ERROR when maxReconnectionAttempts is reached so that sounds like a bug. Would you be able to put instructions to replicate into an issue?
Thanks!